scripts/qa_crowbarsetup.sh

```diff
@@ -3310,11 +3310,7 @@ function oncontroller_run_tempest
         # output the slowest tests from the latest run
         testr slowest
     fi
-    if iscloudver 7plus; then
-        ostestr last --subunit --no-pretty > tempest.subunit.log
```

Comment:
This was done because ostestr generated version 2 of subunit.
And testr generated version 1, but the filter subunit-1to2 was dropped when switching to the single-spec.
See the original PR: https://github.com/SUSE-Cloud/automation/pull/2222/files

Reply:
Hi! Thanks for your comment!

So, I think there are two issues in this code at least.
1. ostestr doesn't have 'last' sub-command. So `ostestr last ..` is invalid.
2. The tempest tests runs already here https://github.com/SUSE-Cloud/automation/pull/2269/files/12f6a3be70522c61f86f09311e684d7f349e1dd9#diff-c202af9895fd7be9224b15a93a00b0ffL3299 . So, it runs the tests again if we do `ostestr last` and `ostestr` here. It just wastes the time and resources.

To resolve these issue, I think there are two options.
1. Use `ostestr` instead of `tempest run` in L.3299 and 3302. Tempest tests should work with `ostestr`, too. https://docs.openstack.org/tempest/latest/overview.html#quickstart
2. Use `stestr` instead of `ostestr` in L.3314 and L3334. But I'm not sure we can use `stestr` in SOC8 or less versions.

So, I think Option-1 fits here if we can't use `stestr`. What do you think?

Reply:
1. Please leave `tempest run` alone :)
2. I didn't know that ostestr does not have a last command. When i checked it was not running all the tests again, but just retrieved the last run results. I will double check today.
3. Whatever you choose make sure you can generate subunit v1 and subunit v2 versions of the last run results. We need that for rally and/or defcore/refstack client.

Reply:
1. OK, I'll leave it :)
2. umm, it's weird. IIRC, ostestr doesn't have like that feature(retrieve the last run results) from the beginning.
3. So, to just convert from subunit v1 streams to v2, I think just using subunit-1to2 is enough. I don't know why we need to drop the `subunit-1to2` command. It should be included in python-subunit. I actually don't know the **single-spec** which you mentioned...

Reply:
After re-read the source code and comments, I got confused.
I think `testr last --subunit` should generate a version 2 [0]. So, if we need version 2 stream, just `testr last --subunit` is enough. And if we need version 1 stream, `testr last --subunit | subunit-2to1` should work.
So, which version do we need here and below? I suppose we want **version 1**. Am I correct?

[0] https://github.com/testing-cabal/testrepository/blob/master/testrepository/commands/last.py#L49